library/std/src/sys/zkvm/abi.rs

```diff
@@ -1,5 +1,94 @@
-extern "Rust" {
-    pub(crate) fn zkvm_abi_alloc_words(nwords: usize) -> *mut u32;
-    pub(crate) fn zkvm_abi_write_stdout(buf: &[u8]);
-    pub(crate) fn zkvm_abi_write_stderr(buf: &[u8]);
+//! ABI definitions for symbols exported by risc0-zkvm-platform.
+
+// Included here so we don't have to depend on risc0-zkvm-platform.
+//
+// TODO: Should we move this to the "libc" crate?  It seems like other
+// architectures put a lot of this kind of stuff there.  But there's
+// currently no risc0 fork of the libc crate, so we'd either have to
+// fork it or upstream it.
+
+pub const DIGEST_WORDS: usize = 8;
+
+/// Standard IO file descriptors for use with sys_read and sys_write.
+pub mod fileno {
+    pub const STDIN: u32 = 0;
+    pub const STDOUT: u32 = 1;
+    pub const STDERR: u32 = 2;
+    pub const JOURNAL: u32 = 3;
+}
+
+extern "C" {
+    // Syscall names are all nul-terminated c-style string.  Raw access to syscalls:
+    pub fn syscall_0(name: *const u8, from_host: *mut u32, from_host_words: usize) -> (u32, u32);
```

Comment:
I'm surprised that we don't mark all these functions as `unsafe`, or does `extern "C"` imply this?